core/src/main/java/hudson/model/UnprotectedRootAction.java

```diff
@@ -32,4 +32,26 @@
  * @since 1.398
  */
 public interface UnprotectedRootAction extends RootAction, ExtensionPoint {
+
+    /**
+     * Gets the URL path name.
+     *
+     * <p>
+     * For example, if this method returns "xyz", and if the parent object
+     * (that this action is associated with) is bound to /foo/bar/zot,
+     * then this action object will be exposed to /foo/bar/zot/xyz.
+     *
+     * <p>
+     * This method should return a string that's unique among other {@link Action}s.
+     *
+     * <p>
+     * <strong>Unlike normal {@link Action}s the returned string is always assumed to be relative to the context path of the Jenkins webapp and should not be prefixed with a '/'.</strong>
+     * 
+     * As UnprotectedRootActions are always accessible this should not return a {@code null} value.
+     * @return
+     *      a non-null url name.
+     */
+    @Override
+    String getUrlName();
```

Comment:
I wasn't convinced that there was not a  case that returning null was acceptable (although the url would be jenkins/null),
however I can also see that as a different bug and that perhaps jenkins should check for a null name on `UnprotectedRootAction` (hence should rather than must).